mesonbuild/mesonlib.py

```diff
@@ -363,6 +363,17 @@ def is_dragonflybsd():
 def is_freebsd():
     return platform.system().lower() == 'freebsd'
 
+# It's possible to run Meson under bash while cross-compiling for windows
+# and so can't assume that is_windows() implies running under cmd. It's
+# conventional for SHELL to be set when running under a unix shell but
+# not under cmd, so (just on windows) we use that to determine the shell
+# type...
+def is_sh_shell():
+    return (not is_windows()) or ('SHELL' in os.environ)
```

Comment:
This function doesn't seem to be used anywhere, so should probably be removed.

It could also be written as 'not is_cmd_shell()'.